### ISSUE TITLE
Remove all content before sync

### ISSUE THREAD

**Issue by tisonkun** [apache/pulsar-site#175]
@urfreespace @Anonymitaet Otherwise, we can suffer from some files deleted from upstream but still remain in this repo.

**Comment by michaeljmarshall** [comment]
I think this kind of change will break certain links that I left in place a couple months ago. For example, many of the SNAPSHOT links here https://pulsar.apache.org/api/client/ will go away. I think it is fine to get rid of them if we are ready to do it, but I was slightly worried that users would have saved links to those endpoints, so I didn't remove them.

**Comment by tisonkun** [comment]
@michaeljmarshall then it's an issue about the source of truth. If we retain those content, how can we remove it from the main repo?

The original purpose of this issue is to avoid dangling content under `docs` folder, I may narrow the scope there, though.

**Comment by tisonkun** [comment]
Since docs are versioned, it should not be a big deal. If we'd like to retain the link, we set up a redirection.

**Comment by michaeljmarshall** [comment]
That is a fair point about the source of truth. Maybe we just need to have a landing page? In the case of the API docs, the old content wasn't even based on released versions of Pulsar, so its value is questionable, anyway.

**Comment by Anonymitaet** [comment]
@tisonkun 
Suggest keeping all docs in the `Pulsar` repo, there was a similar discussion showing the benefits of this way.
https://lists.apache.org/thread/mf2xwntfgn84dq78ksqv22jk3drq6xb3

**Comment by tisonkun** [comment]
@Anonymitaet Thanks for your information.

@michaeljmarshall If I get it right, in the mail thread linked your preference is keeping the "current" (unreleased) document in the main repo? Keep released versioned document, especially still retain those like "version-2.1.0-incubating", "version-2.1.1-incubating" really help?

**Comment by michaeljmarshall** [comment]
> If I get it right, in the mail thread linked your preference is keeping the "current" (unreleased) document in the main repo?

I think it is helpful to keep docs that are manually maintained in the apache/pulsar repo. Once a version is no longer supported, I don't think we need those docs in the apache/pulsar repo since the version itself is not getting updated.

> Keep released versioned document, especially still retain those like "version-2.1.0-incubating", "version-2.1.1-incubating" really help?

I support removing these docs from the apache/pulsar repo, and probably from the website, but I am not sure. One of my biggest problems with the old docs is that I still get served them by search engines above more recent docs. We definitely want users to find recent documentation.

**Comment by tisonkun** [comment]
>  that are manually maintained

Good point. I will come up with a proposal to describe this topic in details. It's actually coupled with how we perform patch releases and how to maintain those versioned docs.
